### ISSUE TITLE
Address ZAP Absence of Anti-CSRF tokens finding (Medium) Due for 06/2022 ConMon

### ISSUE THREAD

**Issue by pburkholder** [cloud-gov/product#1712]
In order to keep our compliance/security ship on course, we want to address Medium findings for Anti-CSRF tokens.

The ZAP project, in https://github.com/zaproxy/zap-extensions/pull/3548, updated the severity of "Absence of Anti-CSRF tokens" from Low to Medium. They also downgraded the confidence of the finding from Medium to Low.

We have some sites that trigger this finding, and since this is now a Medium it's time we dig in and determine
- are these real findings? If not, document how they're false positives (or make PRs to the `zap-extensions` project)
- fix them as needed with
  - PRs to our our projects
  - Issues/PRs to our dependent project

For full accounting of the findings in question, see the [2022 Zap and Nessus findings in Drive](https://drive.google.com/drive/folders/0B5fn0WMJaYDnaFdCak5WNWRGb1U?resourcekey=0-c4qYiNlJzM8vLp1bMifmIg)

## Security considerations
Yes. Addresses scanner findings.

**Comment by pburkholder** [comment]
This seems to be real. Most of the findings are for the `cloud.gov` site, where we have source forms: 

```
 	<form id="search_form" class="usa-search usa-search--small" action="https://search.usa.gov/search" accept-charset="UTF-8" method="get">
```

And then there are the internal sites that redirect to `<form method="GET" action="/oauth2/start">` -- this may already be addressed with oauth proxy updates.

**Comment by pburkholder** [comment]
@mogul - We either need this fixed by 6/20 or carry it as an O.R. I think team is doing CSPs first, and that's OK. Just FYI.

**Comment by pburkholder** [comment]
We also need to address the IDP/login findings:

* URL: https://login.fr.cloud.gov/saml/login/alias/login.fr.cloud.gov?disco=true&idp=https%3A%2F%2Fidp.fr.cloud.gov
   * Evidence: `<form action="https&#x3a;&#x2f;&#x2f;idp.fr.cloud.gov&#x2f;profile&#x2f;SAML2&#x2f;POST&#x2f;SSO" method="post">`
* URL: https://idp.fr.cloud.gov/profile/SAML2/POST/SSO?execution=e2s2
  * Evidence: `<form  action="https&#x3a;&#x2f;&#x2f;login.fr.cloud.gov&#x2f;saml&#x2f;SSO&#x2f;alias&#x2f;login.fr.cloud.gov"  method="post">`

**Comment by pburkholder** [comment]
@rcgottlieb Re-opened for final documentation.

**Comment by rcgottlieb** [comment]
[Documentation for Anti-CSRF tokens findings](https://docs.google.com/document/d/1VYVmCA7VSx578tj_h5SOXgFlOyg6LRTA/edit#heading=h.ed86mh11kw80)

**Comment by pburkholder** [comment]
I need to map this work back to the POA&Ms CG129 and CG364 before we can close it.

**Comment by wz-gsa** [comment]
No longer needed
